core/src/main/scala/kafka/tools/ConsumerPerformance.scala

```diff
@@ -266,11 +280,12 @@ object ConsumerPerformance {
           val messageAndMetadata = iter.next
           messagesRead += 1
           bytesRead += messageAndMetadata.message.length
+          val currentTimeMillis = System.currentTimeMillis
```

Comment:
I have not seen a much better alternative to this. This method's cost is in the order of nanoseconds, which is not terribly bad. And if we are to do the reporting by time intervals we would need to somehow keep track of time. One thing that can be done is swap the two if conditions and move the `System.currentTimeMillis` call inside the outer if. That helps with when showing detailed stats is not requested.

Please let me know if you can think of a better approach.